ydb/library/yql/public/udf/arrow/bit_util.h

```diff
@@ -15,6 +15,35 @@ inline ui8* CompressAsSparseBitmap(const ui8* src, size_t srcOffset, const ui8*
     return dst;
 }
 
+inline ui8* DecompressToSparseBitmap(ui8* dstSparse, const ui8* src, size_t srcOffset, size_t count) {
+    if (srcOffset != 0) {
+        size_t offsetBytes = srcOffset >> 3;
+        size_t offsetTail = (srcOffset & 7);
+        src += offsetBytes;
+        for (ui8 i = offsetTail; i < 8; i++) {
+            *dstSparse++ = (*src >> i) & 1u;
+        }
+        src++;
```

Comment:
But everything is copied in the loop above, since `offsetTail` is `0`. So the only issue is that the "offsetTail" loop can be unrolled for this bitmap byte either.

Anyway, you're right, `srcOffset != 0` should be changed to `srcOffset % 8 != 0`.

Reply:
[Updated](https://github.com/ydb-platform/ydb/pull/3400/files#diff-b8e5dc41387f49b5eb79979cfe9dc8c2cfccf15491225b4801cbdb37fcb778d8R19).